src/common/commonutils/CommandUtils.c

```diff
@@ -35,6 +73,22 @@ int ExecuteCommand(void* context, const char* command, bool replaceEol, bool for
         return E2BIG;
     }
 
+    // Allow mocked call for unit testing of things that execute commands.
+    struct mockCommand* mock = g_mockCommand;
+    while (NULL != mock) {
+        size_t stringLen = strlen(mock->expectedCommand);
+        if (!mock->matchPrefix && (strlen(command) > stringLen))
+        {
+            stringLen = strlen(mock->expectedCommand);
+        }
+        if (0 == strncmp(mock->expectedCommand, command, stringLen))
+        {
+            *textResult = DuplicateString(mock->output);
+            return mock->returnCode;
+        }
+	mock = mock->next;
+    }
```

Comment:
We should not bloat production code with mocking code for unit-tests. We should instead place mocking code in the unit-tests themselves. Can we please do this?

Reply:
No, we can't, as we want to simulate the real ExecuteCommand function and not have to replace it in the tested code.